### PR TITLE
[FLINK-13620][Tests] Update FlinkKafkaProducerMigrationTest to restore from 1.9 savepoint

### PR DESCRIPTION
**Pull request by yanghua**
##What is the purpose of the change

*This pull request updates FlinkKafkaProducerMigrationTest to restore from 1.9 savepoint*

## Brief change log

  - *Generated binary snapshot file*
  - *Updated migration version*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by yanghua** [comment]
cc @tillrohrmann

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerMigrationTest.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
 	@Parameterized.Parameters(name = "Migration Savepoint: {0}")
 	public static Collection<MigrationVersion> parameters() {
 		return Arrays.asList(
-			MigrationVersion.v1_8);
+			MigrationVersion.v1_9);
```
Comment (tillrohrmann) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerMigrationTest.java]: I think we should also keep `MigrationVersion.v1_8` here.

Comment (yanghua) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerMigrationTest.java]: I am not against it. But in many test cases, we updated the migration version. Why not update this?

Comment (tillrohrmann) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerMigrationTest.java]: I think we always added the new version `v1_9` to the set of tested versions if there wasn't an incompatibility. Here we deactivate the test which tests the migration from checkpoints of version `1.8`

Comment (yanghua) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerMigrationTest.java]: OK, I will reset it to `v1_8`.

Comment (yanghua) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerMigrationTest.java]: Done

Comment (tillrohrmann) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerMigrationTest.java]: I think it should be:
```suggestion
                         MigrationVersion.v1_8,
			MigrationVersion.v1_9);
```

**Comment by tillrohrmann** [comment]
@yanghua I think we should add both versions `MigrationVersion.v1_8` and `MigrationVersion.v1_9` to the array list.

**Comment by yanghua** [comment]
@tillrohrmann Sorry, my mistake. I misunderstood your first comment. Have updated the PR.

**Pull request closed without merging**